File: android/app/src/main/java/com/marco/fittracker/ui/PlannerGoal.kt
package com.marco.fittracker.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.FlowRow
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material.icons.filled.Close
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.window.Dialog
import com.marco.fittracker.data.L
import com.marco.fittracker.data.pf
import com.marco.fittracker.data.t
import com.marco.fittracker.data.trimNum
import java.time.LocalDate

@OptIn(androidx.compose.foundation.layout.ExperimentalLayoutApi::class)
@Composable
private fun GoalPillRow(options: List<Pair<String, String>>, selected: String, onSelect: (String) -> Unit) {
    FlowRow(horizontalArrangement = Arrangement.spacedBy(7.dp), verticalArrangement = Arrangement.spacedBy(7.dp)) {
        options.forEach { (value, label) ->
            val on = value == selected
            Text(
                label, color = if (on) T.bg else T.txt, fontSize = 12.sp, fontWeight = FontWeight.SemiBold,
                modifier = Modifier.clip(RoundedCornerShape(10.dp)).background(if (on) T.acc else T.c2)
                    .border(1.dp, if (on) Color.Transparent else T.brd, RoundedCornerShape(10.dp))
                    .clickable { onSelect(value) }.padding(vertical = 9.dp, horizontal = 14.dp)
            )
        }
    }
}

@Composable
private fun GoalField(label: String, value: String, onChange: (String) -> Unit, ph: String) {
    Column(Modifier.fillMaxWidth()) {
        Lbl(label)
        Spacer(Modifier.height(7.dp))
        InputField(value, onChange, ph, KeyboardType.Decimal)
    }
}

// MARK: - Goal editor (always reachable from Home)
@Composable
fun GoalEditorDialog(onClose: () -> Unit) {
    val store = LocalStore.current
    val toast = LocalToast.current
    val p = store.prefs
    var goalMode by remember { mutableStateOf(p.goal.raw) }
    var startW by remember { mutableStateOf(trimNum(p.startWeight)) }
    var goalW by remember { mutableStateOf(trimNum(p.goalWeight)) }
    var goalBF by remember { mutableStateOf(trimNum(p.goalBF)) }
    var rate by remember { mutableStateOf(p.weeklyRate?.let { trimNum(it) } ?: "") }

    Dialog(onDismissRequest = onClose) {
        Column(
            Modifier.fillMaxWidth().heightIn(max = 620.dp).clip(RoundedCornerShape(T.radius))
                .background(T.bg).border(1.dp, T.brd, RoundedCornerShape(T.radius))
                .padding(18.dp).verticalScroll(rememberScrollState())
        ) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(t("goal.title").uppercase(), color = T.txt, fontSize = 18.sp, fontWeight = FontWeight.Bold)
                Spacer(Modifier.width(6.dp))
                InfoButton("goal")
                Spacer(Modifier.weight(1f))
                Icon(Icons.Filled.Close, null, tint = T.sub, modifier = Modifier.size(24.dp).clickable { onClose() })
            }
            Spacer(Modifier.height(12.dp))
            Text(t("goal.hint"), color = T.sub, fontSize = 12.sp, lineHeight = 17.sp)
            Spacer(Modifier.height(14.dp))

            Lbl(t("ob.goal_mode")); Spacer(Modifier.height(8.dp))
            GoalPillRow(listOf("cut" to t("nut.cut"), "maintain" to t("nut.maintain"), "bulk" to t("nut.bulk")), goalMode) { goalMode = it }
            Spacer(Modifier.height(14.dp))
            Row(horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                Box(Modifier.weight(1f)) { GoalField(t("goal.start_weight"), startW, { startW = it }, "88") }
                Box(Modifier.weight(1f)) { GoalField(t("ob.goal_weight"), goalW, { goalW = it }, "80") }
            }
            Spacer(Modifier.height(12.dp))
            Row(horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                Box(Modifier.weight(1f)) { GoalField(t("pc.goal_bf"), goalBF, { goalBF = it }, "15") }
                Box(Modifier.weight(1f)) { GoalField(t("ob.rate"), rate, { rate = it }, "-0.5") }
            }
            Spacer(Modifier.height(18.dp))
            BigButton(t("save")) {
                val np = store.prefs.copy(
                    goalMode = goalMode,
                    startWeight = pf(startW).takeIf { it > 0 } ?: store.prefs.startWeight,
                    goalWeight = pf(goalW).takeIf { it > 0 } ?: store.prefs.goalWeight,
                    goalBF = pf(goalBF).takeIf { it > 0 } ?: store.prefs.goalBF,
                    weeklyRate = rate.ifBlank { null }?.let { pf(it) }
                )
                store.updatePrefs(np)
                toast.show(t("goal.saved"))
                onClose()
            }
        }
    }
}

// MARK: - Weekly planner dialog (assign each weekday a plan / cardio / rest)
@Composable
fun WeeklyPlanDialog(onClose: () -> Unit) {
    val store = LocalStore.current
    val toast = LocalToast.current
    val dayNames = if (L.lang == "en")
        listOf("Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday")
    else listOf("Lunedì", "Martedì", "Mercoledì", "Giovedì", "Venerdì", "Sabato", "Domenica")
    val todayMon = (LocalDate.now().dayOfWeek.value - 1).coerceIn(0, 6)

    Dialog(onDismissRequest = onClose) {
        Column(
            Modifier.fillMaxWidth().heightIn(max = 640.dp).clip(RoundedCornerShape(T.radius))
                .background(T.bg).border(1.dp, T.brd, RoundedCornerShape(T.radius))
                .padding(18.dp).verticalScroll(rememberScrollState())
        ) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(t("plan.week").uppercase(), color = T.txt, fontSize = 18.sp, fontWeight = FontWeight.Bold)
                Spacer(Modifier.width(6.dp))
                InfoButton("weekplan")
                Spacer(Modifier.weight(1f))
                Icon(Icons.Filled.Close, null, tint = T.sub, modifier = Modifier.size(24.dp).clickable { onClose() })
            }
            Spacer(Modifier.height(12.dp))
            Text(t("plan.week_hint"), color = T.sub, fontSize = 12.sp, lineHeight = 17.sp)
            Spacer(Modifier.height(14.dp))

            for (wd in 0 until 7) {
                DayScheduleRow(wd, dayNames[wd], wd == todayMon)
                if (wd < 6) { Spacer(Modifier.height(4.dp)); Box(Modifier.fillMaxWidth().height(1.dp).background(T.brd)); Spacer(Modifier.height(4.dp)) }
            }

            if (store.prefs.hasSchedule) {
                Spacer(Modifier.height(16.dp))
                Box(Modifier.fillMaxWidth().heightIn(min = 46.dp).clip(RoundedCornerShape(T.radiusS))
                    .border(1.dp, T.red.copy(alpha = 0.4f), RoundedCornerShape(T.radiusS))
                    .clickable { store.clearSchedule(); toast.show(t("plan.saved")) }
                    .padding(vertical = 12.dp), contentAlignment = Alignment.Center) {
                    Text(t("plan.clear"), color = T.red, fontSize = 13.sp, fontWeight = FontWeight.SemiBold)
                }
            }
        }
    }
}

@Composable
private fun DayScheduleRow(wd: Int, dayName: String, isToday: Boolean) {
    val store = LocalStore.current
    var open by remember { mutableStateOf(false) }
    val cur = store.prefs.weekSchedule[wd]
    Row(Modifier.fillMaxWidth().padding(vertical = 7.dp), verticalAlignment = Alignment.CenterVertically) {
        Column(Modifier.weight(1f)) {
            Text(dayName, color = if (isToday) T.acc else T.txt, fontSize = 14.sp, fontWeight = FontWeight.SemiBold)
            if (isToday) Text(t("plan.today").uppercase(), color = T.acc, fontSize = 8.sp, fontWeight = FontWeight.SemiBold, letterSpacing = 1.sp)
        }
        Box {
            Row(
                Modifier.clip(RoundedCornerShape(9.dp)).background(T.c2).border(1.dp, T.brd, RoundedCornerShape(9.dp))
                    .clickable { open = true }.padding(vertical = 8.dp, horizontal = 12.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Box(Modifier.size(9.dp).clip(CircleShape).background(slotColor(store, cur)))
                Spacer(Modifier.width(6.dp))
                Text(slotLabel(store, cur), color = T.txt, fontSize = 13.sp, fontWeight = FontWeight.SemiBold)
            }
            DropdownMenu(expanded = open, onDismissRequest = { open = false }) {
                DropdownMenuItem(text = { Text(t("plan.auto")) }, onClick = { store.setSchedule(wd, ""); open = false })
                DropdownMenuItem(text = { Text(t("plan.rest")) }, onClick = { store.setSchedule(wd, "rest"); open = false })
                store.plans.forEach { p ->
                    DropdownMenuItem(text = { Text(p.name) }, onClick = { store.setSchedule(wd, p.id); open = false })
                }
                store.cardioTypes.forEach { c ->
                    DropdownMenuItem(text = { Text(c.name) }, onClick = { store.setSchedule(wd, c.id); open = false })
                }
            }
        }
    }
}

private fun slotLabel(store: com.marco.fittracker.data.Store, id: String): String = when {
    id.isEmpty() -> t("plan.auto")
    id == "rest" -> t("plan.rest")
    else -> store.plans.firstOrNull { it.id == id }?.name
        ?: store.cardioTypes.firstOrNull { it.id == id }?.name ?: t("plan.auto")
}
private fun slotColor(store: com.marco.fittracker.data.Store, id: String): Color = when {
    id.isEmpty() -> T.brd2
    id == "rest" -> T.restFill
    else -> store.plans.firstOrNull { it.id == id }?.let { hexColor(it.color) }
        ?: store.cardioTypes.firstOrNull { it.id == id }?.let { hexColor(it.color) } ?: T.brd2
}

// MARK: - Compact weekly-plan card for Home
@Composable
fun WeeklyPlanCard() {
    val store = LocalStore.current
    var editing by remember { mutableStateOf(false) }
    val sched = store.prefs.weekSchedule
    val todayMon = (LocalDate.now().dayOfWeek.value - 1).coerceIn(0, 6)
    val heads = L.weekHeaders
    Card(modifier = Modifier.clickable { editing = true }) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Lbl(t("plan.week"), T.acc2)
            Spacer(Modifier.width(5.dp))
            InfoButton("weekplan", T.acc2)
            Spacer(Modifier.weight(1f))
            Text((if (store.prefs.hasSchedule) t("plan.scheduled") else t("plan.rotation")).uppercase(),
                color = T.sub, fontSize = 9.sp, fontWeight = FontWeight.SemiBold)
            Icon(Icons.AutoMirrored.Filled.KeyboardArrowRight, null, tint = T.sub, modifier = Modifier.size(16.dp))
        }
        Spacer(Modifier.height(12.dp))
        Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
            for (wd in 0 until 7) {
                Column(Modifier.weight(1f), horizontalAlignment = Alignment.CenterHorizontally) {
                    Text(heads[wd], color = if (wd == todayMon) T.acc else T.sub, fontSize = 9.sp, fontWeight = FontWeight.SemiBold)
                    Spacer(Modifier.height(6.dp))
                    Box(
                        Modifier.fillMaxWidth().height(30.dp).clip(RoundedCornerShape(8.dp))
                            .background(slotChipColor(store, sched[wd]))
                            .border(1.5.dp, if (wd == todayMon) T.acc else Color.Transparent, RoundedCornerShape(8.dp)),
                        contentAlignment = Alignment.Center
                    ) {
                        if (sched[wd] == "rest") Icon(restIcon, null, tint = T.bg, modifier = Modifier.size(13.dp))
                    }
                }
            }
        }
    }
    if (editing) WeeklyPlanDialog { editing = false }
}

private fun slotChipColor(store: com.marco.fittracker.data.Store, id: String): Color = when {
    id.isEmpty() -> T.c3
    id == "rest" -> T.restFill
    else -> store.plans.firstOrNull { it.id == id }?.let { hexColor(it.color) }
        ?: store.cardioTypes.firstOrNull { it.id == id }?.let { hexColor(it.color) } ?: T.c3
}

File: ios/FitTracker/Views/PlannerGoal.swift
import SwiftUI

// MARK: - Goal editor (always reachable from Home)
// The goal is fixed at onboarding and only ever changes here, on purpose — it
// must not drift with daily check-ins. Start weight is the baseline progress is
// measured from, so it's editable too.
struct GoalEditorView: View {
    @EnvironmentObject var store: Store
    @EnvironmentObject var toast: ToastCenter
    @Environment(\.dismiss) private var dismiss

    @State private var goalWeight = ""
    @State private var startWeight = ""
    @State private var goalBF = ""
    @State private var goalMode = "maintain"
    @State private var rate = ""

    var body: some View {
        ZStack {
            Theme.bg.ignoresSafeArea()
            ScrollView {
                VStack(alignment: .leading, spacing: 16) {
                    HStack {
                        Text(t("goal.title").uppercased()).font(.head(18, .bold)).tracking(1).foregroundColor(Theme.txt)
                        InfoButton(id: "goal")
                        Spacer()
                        Button { tap(); dismiss() } label: {
                            Image(systemName: "xmark").foregroundColor(Theme.sub).frame(width: 34, height: 34)
                        }
                    }
                    .padding(.top, 18)

                    Text(t("goal.hint")).font(.system(size: 12)).foregroundColor(Theme.sub).lineSpacing(3)

                    Card {
                        FieldRow(label: t("ob.goal_mode")) {
                            PillSelect(options: ["cut", "maintain", "bulk"], title: goalTitleLocal, selection: $goalMode)
                        }
                        Spacer().frame(height: 14)
                        HStack(spacing: 12) {
                            FieldRow(label: t("goal.start_weight")) { InputField(placeholder: "88", text: $startWeight) }
                            FieldRow(label: t("ob.goal_weight")) { InputField(placeholder: "80", text: $goalWeight) }
                        }
                        Spacer().frame(height: 14)
                        HStack(spacing: 12) {
                            FieldRow(label: t("pc.goal_bf")) { InputField(placeholder: "15", text: $goalBF) }
                            FieldRow(label: t("ob.rate")) { InputField(placeholder: "-0.5", text: $rate) }
                        }
                    }

                    BigButton(title: t("save")) { save() }
                        .padding(.bottom, 30)
                }
                .padding(.horizontal, 18)
            }
            .scrollDismissesKeyboard(.interactively)
        }
        .preferredColorScheme(.dark)
        .onAppear {
            let p = store.prefs
            goalWeight = trimNum(p.goalWeight)
            startWeight = trimNum(p.startWeight)
            goalBF = trimNum(p.goalBF)
            goalMode = p.goal.rawValue
            rate = p.weeklyRate.map { trimNum($0) } ?? ""
        }
    }

    private func goalTitleLocal(_ v: String) -> String {
        switch v { case "cut": return t("nut.cut"); case "bulk": return t("nut.bulk"); default: return t("nut.maintain") }
    }

    private func save() {
        var p = store.prefs
        if pf(goalWeight) > 0 { p.goalWeight = pf(goalWeight) }
        if pf(startWeight) > 0 { p.startWeight = pf(startWeight) }
        if pf(goalBF) > 0 { p.goalBF = pf(goalBF) }
        p.goalMode = goalMode
        p.weeklyRate = rate.isEmpty ? nil : pf(rate)
        store.prefs = p
        haptic(.success)
        toast.show(t("goal.saved"))
        dismiss()
    }
}

// MARK: - Weekly planner (assign each weekday a plan / cardio / rest)
struct WeeklyPlanView: View {
    @EnvironmentObject var store: Store
    @EnvironmentObject var toast: ToastCenter
    @Environment(\.dismiss) private var dismiss

    // Monday-first labels independent of locale weekday order.
    private var dayNames: [String] {
        L.lang == "en"
            ? ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
            : ["Lunedì", "Martedì", "Mercoledì", "Giovedì", "Venerdì", "Sabato", "Domenica"]
    }

    var body: some View {
        let todayMon = (Calendar.current.component(.weekday, from: Date()) + 5) % 7
        ZStack {
            Theme.bg.ignoresSafeArea()
            ScrollView {
                VStack(alignment: .leading, spacing: 14) {
                    HStack {
                        Text(t("plan.week").uppercased()).font(.head(18, .bold)).tracking(1).foregroundColor(Theme.txt)
                        InfoButton(id: "weekplan")
                        Spacer()
                        Button { tap(); dismiss() } label: {
                            Image(systemName: "xmark").foregroundColor(Theme.sub).frame(width: 34, height: 34)
                        }
                    }
                    .padding(.top, 18)

                    Text(t("plan.week_hint")).font(.system(size: 12)).foregroundColor(Theme.sub).lineSpacing(3)

                    Card {
                        ForEach(0..<7, id: \.self) { wd in
                            dayRow(wd, isToday: wd == todayMon)
                            if wd < 6 { Rectangle().fill(Theme.brd).frame(height: 1).padding(.vertical, 2) }
                        }
                    }

                    if store.prefs.hasSchedule {
                        Button { tap(); store.clearSchedule(); toast.show(t("plan.saved")) } label: {
                            Text(t("plan.clear")).font(.head(13, .semibold)).foregroundColor(Theme.red)
                                .frame(maxWidth: .infinity, minHeight: 46)
                                .overlay(RoundedRectangle(cornerRadius: Theme.radiusS, style: .continuous).stroke(Theme.red.opacity(0.4), lineWidth: 1))
                        }
                    }
                    Spacer(minLength: 30)
                }
                .padding(.horizontal, 18)
            }
        }
        .preferredColorScheme(.dark)
    }

    private func dayRow(_ wd: Int, isToday: Bool) -> some View {
        let cur = store.prefs.weekSchedule[wd]
        return HStack {
            VStack(alignment: .leading, spacing: 2) {
                Text(dayNames[wd]).font(.system(size: 14, weight: .semibold)).foregroundColor(isToday ? Theme.acc : Theme.txt)
                if isToday { Text(t("plan.today").uppercased()).font(.head(8, .semibold)).tracking(1).foregroundColor(Theme.acc) }
            }
            Spacer()
            Menu {
                Button(t("plan.auto")) { tap(); store.setSchedule(weekday: wd, id: "") }
                Button(t("plan.rest")) { tap(); store.setSchedule(weekday: wd, id: "rest") }
                if !store.plans.isEmpty {
                    Divider()
                    ForEach(store.plans) { p in
                        Button(p.name) { tap(); store.setSchedule(weekday: wd, id: p.id) }
                    }
                }
                if !store.cardioTypes.isEmpty {
                    Divider()
                    ForEach(store.cardioTypes) { c in
                        Button(c.name) { tap(); store.setSchedule(weekday: wd, id: c.id) }
                    }
                }
            } label: {
                HStack(spacing: 7) {
                    ZStack {
                        RoundedRectangle(cornerRadius: 5, style: .continuous).fill(slotColor(cur)).frame(width: 18, height: 18)
                        if cur == "rest" {
                            Image(systemName: Theme.restIcon).font(.system(size: 9, weight: .bold)).foregroundColor(Theme.bg)
                        } else if let icon = slotIcon(cur) {
                            Image(systemName: icon).font(.system(size: 9, weight: .bold)).foregroundColor(Theme.bg)
                        }
                    }
                    Text(slotLabel(cur)).font(.system(size: 13, weight: .semibold)).foregroundColor(Theme.txt)
                    Image(systemName: "chevron.down").font(.system(size: 9)).foregroundColor(Theme.sub)
                }
                .padding(.vertical, 8).padding(.horizontal, 12)
                .background(Theme.c2).clipShape(RoundedRectangle(cornerRadius: 9, style: .continuous))
                .overlay(RoundedRectangle(cornerRadius: 9, style: .continuous).stroke(Theme.brd, lineWidth: 1))
            }
        }
        .padding(.vertical, 7)
    }

    private func slotLabel(_ id: String) -> String {
        if id.isEmpty { return t("plan.auto") }
        if id == "rest" { return t("plan.rest") }
        if let p = store.plans.first(where: { $0.id == id }) { return p.name }
        if let c = store.cardioTypes.first(where: { $0.id == id }) { return c.name }
        return t("plan.auto")
    }
    private func slotColor(_ id: String) -> Color {
        if id.isEmpty { return Theme.brd2 }
        if id == "rest" { return Theme.restFill }
        if let p = store.plans.first(where: { $0.id == id }) { return Color(hex: p.color) }
        if let c = store.cardioTypes.first(where: { $0.id == id }) { return Color(hex: c.color) }
        return Theme.brd2
    }
    private func slotIcon(_ id: String) -> String? {
        if id.isEmpty || id == "rest" { return nil }
        if store.plans.contains(where: { $0.id == id }) { return "dumbbell.fill" }
        if let c = store.cardioTypes.first(where: { $0.id == id }) { return c.sportType.icon }
        return nil
    }
}

// MARK: - Compact weekly-plan card for Home
struct WeeklyPlanCard: View {
    @EnvironmentObject var store: Store
    @State private var editing = false

    var body: some View {
        let sched = store.prefs.weekSchedule
        let todayMon = (Calendar.current.component(.weekday, from: Date()) + 5) % 7
        let heads = L.weekHeaders   // Monday-first single letters
        Button { tap(); editing = true } label: {
            Card {
                HStack(spacing: 2) {
                    Lbl(text: t("plan.week"), color: Theme.acc2)
                    InfoButton(id: "weekplan", color: Theme.acc2)
                    Spacer()
                    Text((store.prefs.hasSchedule ? t("plan.scheduled") : t("plan.rotation")).uppercased())
                        .font(.head(9, .semibold)).tracking(0.8).foregroundColor(Theme.sub)
                    Image(systemName: "chevron.right").font(.system(size: 11)).foregroundColor(Theme.sub).padding(.leading, 4)
                }
                .padding(.bottom, 12)
                HStack(spacing: 6) {
                    ForEach(0..<7, id: \.self) { wd in
                        VStack(spacing: 6) {
                            Text(heads[wd]).font(.head(9, .semibold)).foregroundColor(wd == todayMon ? Theme.acc : Theme.sub)
                            RoundedRectangle(cornerRadius: 8, style: .continuous)
                                .fill(slotColor(sched[wd]))
                                .frame(height: 30)
                                .overlay(RoundedRectangle(cornerRadius: 8, style: .continuous)
                                    .stroke(wd == todayMon ? Theme.acc : Color.clear, lineWidth: 1.5))
                                .overlay {
                                    // Colors alone can collide (e.g. swimming and a
                                    // pull day sharing a hue); show the activity icon
                                    // too, exactly like rest days show a moon.
                                    if sched[wd] == "rest" { RestChip(size: 11) }
                                    else if let icon = slotIcon(sched[wd]) {
                                        Image(systemName: icon).font(.system(size: 13, weight: .bold))
                                            .foregroundColor(Theme.bg)
                                    }
                                }
                        }
                        .frame(maxWidth: .infinity)
                    }
                }
            }
        }
        .buttonStyle(.plain)
        .sheet(isPresented: $editing) { WeeklyPlanView() }
    }

    private func slotColor(_ id: String) -> Color {
        if id.isEmpty { return Theme.c3 }
        if id == "rest" { return Theme.restFill }
        if let p = store.plans.first(where: { $0.id == id }) { return Color(hex: p.color) }
        if let c = store.cardioTypes.first(where: { $0.id == id }) { return Color(hex: c.color) }
        return Theme.c3
    }
    /// SF Symbol for an assigned slot: a dumbbell for strength plans, the sport
    /// icon for cardio activities; nil for empty / rest (handled separately).
    private func slotIcon(_ id: String) -> String? {
        if id.isEmpty || id == "rest" { return nil }
        if store.plans.contains(where: { $0.id == id }) { return "dumbbell.fill" }
        if let c = store.cardioTypes.first(where: { $0.id == id }) { return c.sportType.icon }
        return nil
    }
}
